src/structs/from_iter.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{Generator, GeneratorResult, ValueResult};
+
+/// Creates a generator that wraps an `Iterator`.
+///
```

Comment:
Could you add a comment that it is probably more performant to use a "targeted" generator instead of wrapping an iterator?